Docs: Add ancestor property to block-registration.md doc (#45832)
WordPress/gutenberg: docs/reference-guides/block-api/block-registration.md

Setting `parent` lets a block require that it is only available when nested within the specified blocks.

```js

> // Only allow this block when it is nested in a Columns block

Reviewer: ```suggestion
// Only allow this block when it is a direct child of the Columns block.
```

parent: [ 'core/columns' ],
```

#### ancestor (optional)

-   **Type:** `Array`

> The `ancestor` property makes a block available inside the specified block types at any position of the ancestor block subtree. That allows, for example, to place a ‘Comment Content’ block inside a ‘Column’ block, as long as ‘Column’ is somewhere within a ‘Comment Template’ block. In comparrison to the `parent` property blocks that specify their `ancestor` can be placed anywhere in the subtree whilst blocks with a specified `parent` need to be direct children.

Reviewer 2: should be comparison , not comparrison.

Reviewer: Fixed.

> The `ancestor` property makes a block available inside the specified block types at any position of the ancestor block subtree. That allows, for example, to place a ‘Comment Content’ block inside a ‘Column’ block, as long as ‘Column’ is somewhere within a ‘Comment Template’ block. In comparrison to the `parent` property blocks that specify their `ancestor` can be placed anywhere in the subtree whilst blocks with a specified `parent` need to be direct children.

Reviewer: ```suggestion
The `ancestor` property makes a block available inside the specified block types at any position of the ancestor block subtree. That allows, for example, to place a 'Comment Content' block inside a 'Column' block, as long as 'Column' is somewhere within a 'Comment Template' block. In comparison to the `parent` property blocks that specify their `ancestor` can be placed anywhere in the subtree whilst blocks with a specified `parent` need to be direct children.
```

```js

> // Only allow this block when it is nested in a Columns block

Reviewer: ```suggestion
// Only allow this block when it is nested at any level in a Columns block.
```

> parent: [ 'core/columns' ],

Reviewer 2: The example that is immediately below your addition should be ancestor, not parent; since the code example that you added at line 256 is for the parent attribute; not the ancestor attribute.

Reviewer: Fixed.

> parent: [ 'core/columns' ],

Reviewer: ```suggestion
ancestor: [ 'core/columns' ],
```
